File: frontend/lib/homeowner/types/project.ts
/** Project domain types (homeowner view) — aligned with backend projects/project_tasks tables. */

import type { ProjectStatus, ServiceCategorySlug } from "@/lib/types/domain";

export type { ProjectStatus, ServiceCategorySlug };

export interface Project {
  id: string;
  title: string;
  category: ServiceCategorySlug;
  description: string;
  status: ProjectStatus;
  budgetMin: number;
  budgetMax: number;
  location: string;
  /** 0–100. Derived from completed project_tasks. */
  progress: number;
  quotesCount: number;
  unreadMessages: number;
  coverImageUrl: string | null;
  createdAt: string;
  updatedAt: string;
}

File: bestbuild-api/app/homeowner/schemas/project.py
"""Project schemas — mirror frontend/lib/types/project.ts."""

from typing import Literal

from app.shared.schemas.user import CamelModel

ProjectStatus = Literal[
    "draft",
    "pending",
    "matching",
    "quoted",
    "in_progress",
    "delayed",
    "completed",
    "cancelled",
]


class ProjectOut(CamelModel):
    id: str
    title: str
    category: str
    description: str
    status: ProjectStatus
    budget_min: int
    budget_max: int
    location: str
    progress: int
    quotes_count: int
    unread_messages: int
    cover_image_url: str | None = None
    created_at: str
    updated_at: str


class ProjectCreate(CamelModel):
    """Placeholder request model for POST /projects (project intake form)."""

    title: str
    category: str
    description: str
    budget_min: int
    budget_max: int
    location: str
